vulcanai/utils.py

```diff
@@ -419,3 +430,163 @@ def stitch_datasets(df_list, on, index_list=None):
     merged_df.fillna(0, inplace=True)
 
     return merged_df
+
+
+def feature_directionality(features, data, val_x, val_y, dense_net, filename):
+    """
+    Will conduct tests to figure out directionality of features by finding all the unique feature values present in the
+    dataset and setting the feature for all examples to every unique value of that feature.
+
+    Args:
+        features: list of all features in the model
+        data: dataframe with the data, features, y values
+        val_x: the features of the examples in the validation set
+        vay_y: the labels of the examples in the validation set
+        filename: the name of the file to save the test results to
+    """
+    test_df = None
+    for j in range(len(features)):
+        feature_val_list = data[features[j]]
+        unique_feature_values = list(np.unique(feature_val_list))
+        val_x_temp = copy.deepcopy(val_x)
+        for i in range(len(unique_feature_values)):
+            val_x_temp[:, j] = unique_feature_values[i]
+            print("***************************")
+            print("Testing value " + str(unique_feature_values[i]) + " for feature " + str(features[j]))
+            test_df = run_and_save_test(network=dense_net, test_x=val_x_temp, test_y=val_y, test_df=test_df,
+                                        feature=features[j], value=unique_feature_values[i])
+    test_df.to_csv(filename + ".csv")
+
+
+def run_and_save_test(network, test_x, test_y, test_df, feature, value):
+    """
+    Will conduct the test suite to determine model strength and return an updated dataframe test results.
+
+    Args:
+        test_x: data the model has not yet seen to predict
+        test_y: corresponding truth vectors
+        test_df: dataframe with aggregated test results
+        feature: the feature that is being variesd
+        value: value of the feature that is being tested
+    """
+
+    temp_df = pd.DataFrame(columns=["Feature", "Value", "TP", "FP", "TN", "FN",
+                                        "Accuracy", "Sensitivity", "Specificity", "PPV",
+                                        "NPV", "AUC"])
+
+    if network.num_classes is None or network.num_classes == 0:
+        raise ValueError('There\'s no classification layer')
+
+    if test_y.shape[1] > 1:
+        test_y = get_class(test_y)  # Y is in one hot representation
+
+    raw_prediction = network.forward_pass(input_data=test_x,
```

Comment:
similar to my last comment from a previous PR, is there no way to adapt the use of run_test? a lot of this code is duplicated from there